gdc_ng_models/models/studyrule.py

```diff
@@ -0,0 +1,137 @@
+"""Data models to describe studies.
+
+Studies will be used for the single-study controlled-access feature in gdcapi.
+These models provide a way to associate programs and projects to studies.
+"""
+from sqlalchemy import orm
+from sqlalchemy.ext import declarative
+from sqlalchemy.sql import schema, sqltypes
+
+from gdc_ng_models.models import audit
+
+Base = declarative.declarative_base()
+
+
+class StudyRule(Base, audit.AuditColumnsMixin):
+    """A study rule for use with single-study controlled-access.
+
+    Attributes:
+        id: A unique identifier for the study rule.
+        name: A unique, informational name for the study rule.
+        created_date: The date and time when the record is created.
+        updated_date: The date and time when the record is updated.
+    """
+
+    __tablename__ = "study_rule"
+    id_seq = schema.Sequence(name="study_rule_id_seq", metadata=Base.metadata)
+    id = schema.Column(sqltypes.Integer, nullable=False, server_default=id_seq.next_value())
+    name = schema.Column(sqltypes.Text, nullable=False)
+
+    __table_args__ = (
+        schema.PrimaryKeyConstraint("id", name="study_rule_pk"),
+        schema.Index("study_rule_name_idx", "name", unique=True),
+    )
+
+    whole_programs = orm.relationship("StudyRuleProgram", lazy="joined")
+    partial_programs = orm.relationship("StudyRuleProgramProject", lazy="joined")
+
+    def __repr__(self):
+        return "<StudyRule(id={id}, name='{name}', created_date={created_date}, updated_date={updated_date})>".format(
+            id=self.id,
+            name=self.name,
+            created_date=self.created_date.isoformat(),
+            updated_date=self.updated_date.isoformat(),
```

Comment:
In all of these cases, I need to check for the existence for the datetime object. otherwise, these calls to isoformat can fail.